src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -208,6 +208,8 @@ internal static TarEntryType GetCorrectTypeFlagForFormat(TarEntryFormat format,
             T value = T.Zero;
             foreach (byte b in buffer)
             {
+                if (b == (byte)' ') continue; // skip space
```

Comment:
A few notes:

- You could avoid the cast by using 0x20 (or 32 in decimal). The comment can stay since it clarifies it's a space.
- We should add a note explaining why space can be allowed here (some exotic archives might contain entries where the octal value contains unexpected spaces).
- In the tar archives that failed with this error, did you notice that the space was in the middle of the octal number? If not, maybe spaces are only fine to be detected in the edges of the span. I'm conflicted because this could mean complicating the method. What do you think?
- Which octal field threw this exception? I think we should only limit allowing spaces to the fields that failed. I do not like the idea of allowing spaces in all octal fields. But will await for your confirmation.

Reply:
`(byte)' '` cast incurs no cost. In fact, it is the preferred style used extensively in this repo for readability (see e.g. two lines below in the same file).

In `main` branch, we are already ignoring trailing nulls and spaces in all octal fields. while other implementations ignore leading and trailing spaces. I have not find any implementation which is allowing spaces in the middle of the value, so I agree that we should do something about it.

Given:
```
00 004400
00 0abc004400
```

`lib-archive` octal parser parses 44 (octal 36) for both, while busybox tar parses first one as 44 and second one as 0. Note that it applies to _all_ header attributes in both implementations.

I am leaning to follow lib-archive's behavior rather than busybox's, what do you think?

Reply:
`(byte)' '` cast incurs no cost. In fact, it is the preferred style used extensively in this repo for readability (see e.g. two lines below in the same file).

In `main` branch, we are already ignoring trailing nulls and spaces in all octal fields. while other implementations ignore both the leading and trailing spaces. I have not found any implementation which is allowing spaces in the middle of the value, so I agree that we should do something about it.

Given:
```
00 004400
00 0abc004400
```

`lib-archive` octal parser parses 44 (octal 36) for both, while busybox tar parses first one as 44 and second one as 0. Note that it applies to _all_ header attributes in both implementations.

I am leaning to follow lib-archive's behavior rather than busybox's, what do you think?

Reply:
@carlossanlop, I went ahead and made the change.

> what do you think?

From other comments, I assume your reply is you don't like libarchive's behavior, neither busybox's nor GNU tar's. Could you point me to the spec which suggests that we should only trim the space and null from the end of octal values, but not the beginning? i.e. why is this acceptable in the `main` branch https://github.com/dotnet/runtime/blob/8b25e317acace221a647d33867c876fc6e44886d/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs#L205
but not when these bytes appear at the beginning?

Reply:
I think those tar implementations are way too flexible, they are basically ignoring what the spec describes.

Reply:
Can you point me to the part of spec which we are following here?

Reply:
I consulted these sources:

[https://www.freebsd.org/cgi/man.cgi?tar(5)](https://www.freebsd.org/cgi/man.cgi?tar(5))
https://www.gnu.org/software/tar/manual/html_node/tar_toc.html#SEC_Contents
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/pax.html
https://www.ibm.com/docs/en/zos/2.3.0?topic=descriptions-pax-interchange-portable-archives